### ISSUE TITLE
Problem with publish in BlockingConnection

### ISSUE THREAD

**Issue by MarcSky** [pika/pika#1169]
Hey. I have a great application where users constantly create tasks through pika. To the beginning of the BlockingConnection, after users have added a task and  closing connection. Could it be that while one user keeps the connection another user is trying to create it, but since it is blocked, it cannot create one or pika creates a new connection for each task.

The second question is, is it possible to somehow optimize this case, for example, not to close the connection and use the previously created by another user?

Or Maybe I need create multiple consumers?

My problem is that a task is not always created when several users create tasks in parallel. Thank.
```
def create_message(msg):
    params = pika.ConnectionParameters('127.0.0.1')
    connection = pika.BlockingConnection(params)
    channel = connection.channel()
    channel.exchange_declare(
        exchange=MY_EXCHANGE_NAME,
        exchange_type='direct'
    )
    channel.queue_declare(queue=MY_QUEUE_NAME, durable=True)
    channel.basic_publish(
        exchange=MY_EXCHANGE_NAME,
        routing_key=MY_QUEUE_NAME,
        body=pickle.dumps(msg),
        properties=pika.BasicProperties(
            delivery_mode=2,
        )
    )
    connection.close()

```

**Comment by lukebakken** [comment]
Thank you for using Pika.

When you click "New Issue", there is text that people are expected to read, like this:

> GitHub issues are **strictly** used for actionable work and pull
requests.

Please direct your question to either the `pika-python` or `rabbitmq-users` mailing list:

* https://groups.google.com/forum/#!forum/pika-python
* https://groups.google.com/forum/#!forum/rabbitmq-users
